src/Textarea.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import cx from 'classnames';
+import autosize from 'autosize';
+
+/**
+ * @module Textarea
+ */
+class Textarea extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onChange(e) {
+		const { onChange } = this.props;
+		if (onChange) {
+			onChange(e);
+		}
+		this.setState({
+			value: e.target.value,
+		});
+	}
+
+	componentDidMount() {
+		if (this.props.rows === 'auto'){
+			autosize(this.textarea);
+		}
+	}
+
+	render() {
+		const {
+			name,
+			value,	// eslint-disable-line no-unused-vars
+			label,
+			labelClassName,
+			className,
+			error,
+			required,
+			rows,
+			style={},
+			maxHeight,
+			minHeight,
+			id,
+			...other
+		} = this.props;
+
+		const classNames = {
+			textarea: cx(
+				{
+					'field--error': error,
+					'textarea--autoheight': rows === 'auto'
+				},
+				className
+			),
+			label: cx(
+				{ required },
+				labelClassName
+			)
+		};
+
+		const heightConstraints = {
+			minHeight: minHeight,
+			maxHeight: maxHeight
+		};
+
+		return (
+			<div>
+				<label className={classNames.label} htmlFor={id}>
+					{label}
+				</label>
+				<textarea type='text'
+					name={name}
+					required={required}
+					className={classNames.textarea}
+					onChange={this.onChange}
+					rows={rows == 'auto' ? 1 : rows}
+					ref={(textarea) => {this.textarea = textarea;}}
+					style={{ ...style, ...heightConstraints }}
+					value={this.state.value}
+					id={id}
+					{...other}
+				/>
+
+				{ this.props.maxLength && <p className='text--caption align--right'>{this.state.value.length} / {this.props.maxLength}</p> }
```

Comment:
sorry nitpick can you break this up a into multiple lines, getting a bit hard to read

```
{ this.props.maxLength &&
   <p className='text--caption align--right'>
      {this.state.value.length} / {this.props.maxLength}
   </p>
}
```